tools/castle-editor/code/framedesign.pas

```diff
@@ -1970,6 +2027,39 @@ procedure TDesignFrame.ControlsTreeKeyUp(Sender: TObject; var Key: Word;
       ControlsTree.Selected.EditText;
 end;
 
+procedure TDesignFrame.ControlsTreeMouseUp(Sender: TObject;
+  Button: TLclMouseButton; Shift: TShiftState; X, Y: Integer);
+var
+  Sel: TComponent;
+begin
+  if (Button = TLclMouseButton.mbRight) then
+  begin
+    Sel := SelectedComponent;
+    MenuTreeViewItemDuplicate.Enabled := Sel <> nil;
+    MenuTreeViewItemCopy.Enabled := Sel <> nil;
+    MenuTreeViewItemDelete.Enabled := Sel <> nil;
+    if (Sel is TCastleUserInterface) or ((Sel = nil) and (DesignRoot is TCastleUserInterface)) then
+    begin
+      MenuTreeViewItemAddUserInterface.Visible := true;
```

Comment:
Best to toggle both `Enabled` and `Visible`, using `SetEnabledExists` routine.

Feel free to rename `SetEnabledExists` to `SetEnabledVisible` BTW if you consider it better, it seems better to me now :)

Reason: At least in case of other LCL controls, making control invisible is independent from making it disabled. Invisible-but-enabled control can still be activated by a key shortcut.

Reply:
Done in f578d4f183c8b3b97d94e2f49401cfae4d4b67cb